core/templates/components/forms/custom-forms-directives/thumbnail-uploader.component.ts

```diff
@@ -0,0 +1,261 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for uploading images.
+ */
+
+import { OnInit } from '@angular/core';
+import { OnChanges, SimpleChanges } from '@angular/core';
+import { Component, Input } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { AlertsService } from 'services/alerts.service';
+import { ContextService } from 'services/context.service';
+import { CsrfTokenService } from 'services/csrf-token.service';
+import { ImageLocalStorageService } from 'services/image-local-storage.service';
+import { ImageUploadHelperService } from 'services/image-upload-helper.service';
+import { EditThumbnailModalComponent } from './edit-thumbnail-modal.component';
+
+@Component({
+  selector: 'oppia-thumbnail-uploader',
+  templateUrl: './thumbnail-uploader.component.html'
+})
+export class ThumbnailUploaderComponent implements OnInit, OnChanges {
+  @Input() disabled: boolean;
+  @Input() useLocalStorage: boolean;
+  @Input() allowedBgColors: string[];
+  @Input() aspectRatio: string;
+  @Input() bgColor: string;
+  @Input() filename: string;
+  @Input() previewDescription: string;
+  @Input() previewDescriptionBgColor: string;
+  @Input() previewFooter: string;
+  @Input() previewTitle: string;
+  @Input() updateBgColor;
+  @Input() updateFilename;
```

Comment:
please add data types? If these are functions passed from parent component / directive. Then please use @Output here instead.

Reply:
Idk how this wen unnoticed but fixed them now!Thanks for catching this.